scipy/stats/tests/test_binned_statistic.py

```diff
@@ -13,13 +14,13 @@ class TestBinnedStatistic(object):
 
     @classmethod
     def setup_class(cls):
-        np.random.seed(9865)
-        cls.x = np.random.random(100)
-        cls.y = np.random.random(100)
-        cls.v = np.random.random(100)
-        cls.X = np.random.random((100, 3))
-        cls.w = np.random.random(100)
-        cls.u = np.random.random(100) + 1e6
+        rng = check_random_state(9865)
```

Comment:
does this preserve the same rng stream? or might the randomness be different depending on the numpy in the env? I see that the seed is the same, I'm not familiar w/the spec(s) on new numpy random and if the generator would produce same stream or not. Not a blocker provided all tests are passing (they do for me locally) but wanted to clarify.

Reply:
It will generate the same rng stream. The previous code set the global state of the numpy random number generator, whereas these changes create a `RandomState` instance that is used solely for this test class. This means that the test should use exactly the same set of random numbers everytime because it eliminates the possibility that other tests (possibly run in parallel) can affect the state of the global generator.

Reply:
BTW the random number streams from Generator and RandomState are different, even if the same seed is used.

Reply:
ah I think I got it, stream will change because seed/state setting goes from global to local, but two streams w/same seed should generate the same seed. I'm mostly wondering in case we start seeing some spurious test failures on travis w.r.t. tests in `stats` using random numbers. For now it seems ok so no changes needed here. Thanks for clarifying.

Reply:
When I was running the test suite locally this particular test would fail about half the time, even though I hadn't touched it's code.

Reply:
ah ok, just curious if it has anything to do with the standard deviation calculatio/stability test? IIRC I set the exponent, `+ 1e6` so that only bits around 4 to 6 decimals in would lose stability, with a different stream that might change. If that was the test that was failing you could bump to `+ 1e7` and might make the numeric error more stable. If it wasn't that one then we'd need to look into it further if still failing on travis.